projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -87,10 +90,18 @@ export class AuthService {
    */
   async loginWithCredentials(userId: string, password: string): Promise<void> {
     try {
+      let uId: string = userId;
+
+      if (this.authMultisiteIsolationService?.getBaseSiteDecorator()) {
+        uId =
+          userId + this.authMultisiteIsolationService?.getBaseSiteDecorator();
+      }
```

Comment:
It's great we moved a lot of logic to a separate service! This avoids unnecessary responsibilities in the AuthService.

IMHO the implementation detail of building the uid (that it's concatenated _in this order_: userId+decorator and e.g. not the other way around) would be good to also move to the `authMultisiteIsolationService` service.

What about having a method:
```ts
this.authMultisiteIsolationService.getUserId(userId);
//or
this.authMultisiteIsolationService.decorateUserId(userId);
```
?

Reply:
Cool, I added separated method.